src/components/DrawerNavigation/index.tsx

```diff
@@ -0,0 +1,32 @@
+import {MainScreen} from '@screens/MainScreen';
+import {createDrawerNavigator} from '@react-navigation/drawer';
+import {DailyTasksScreen} from '@screens/DailyTasksScreen';
+import {DoneTasksScreen} from '@screens/DoneTasksScreen';
+import {ImportantTasksScreen} from '@screens/ImportantTasksScreen';
+import {DrawerMenu} from '../DrawerMenu';
+
+import {NotesProvider} from '@context/contextProvider';
+import {CategoryScreen} from '@screens/CategoryScreen';
+import {RootStackParamList} from '@customTypes/navigation';
+
+const Drawer = createDrawerNavigator<RootStackParamList>();
+
+export function DrawerNavigation() {
+  return (
+    <NotesProvider>
+      <Drawer.Navigator
+        initialRouteName="MainScreen"
+        screenOptions={{headerShown: false}}
+        drawerContent={props => <DrawerMenu {...props} />}>
+        <Drawer.Screen
+          name="ImportantTasksScreen"
+          component={ImportantTasksScreen}
+        />
+        <Drawer.Screen name="DoneTasksScreen" component={DoneTasksScreen} />
+        <Drawer.Screen name="DailyTasksScreen" component={DailyTasksScreen} />
+        <Drawer.Screen name="MainScreen" component={MainScreen} />
+        <Drawer.Screen name="CategoryScreen" component={CategoryScreen} />
+      </Drawer.Navigator>
+    </NotesProvider>
+  );
```

Comment:
name to enum. This is make a you life easy when you try to navigate to another screen.